modules/packages/UnitTest.chpl

```diff
@@ -441,23 +441,28 @@ module UnitTest {
       array1: The first array to compare.
       array2: The second array to compare.
     */
-    proc assertArrayEqual(array1, array2) throws {
-      type firstType = array1.type,
-          secondType = array2.type;
-      if firstType == secondType {
-        if array1.rank == 1 {
-          assertSequenceEqual(array1, array2, "Array");
-        }
-        else {
-          if !array1.equals(array2) {
-            var tmpString = "assert failed - \n'" + stringify(array1) +"'\n!=\n'"+stringify(array2)+"'";
-            throw new owned AssertionError(tmpString);
-          }
-        }
+    proc assertArrayEqual(array1: [], array2: []) throws {
+      const genericErrorMsg = "assert failed -\n'" + stringify(array1) +
+                              "'\nand\n'"+stringify(array2) + "'\n";
+
+      // Compare array types, size, and shape
+      if array1.type != array2.type {
+        const errorMsg = genericErrorMsg + "are not of same type";
+        throw new owned AssertionError(errorMsg);
+      } else if array1.size != array2.size {
+        const errorMsg = genericErrorMsg + "are not of same size";
+        throw new owned AssertionError(errorMsg);
+      } else if array1.shape != array2.shape {
+        const errorMsg = genericErrorMsg + "are not of same shape";
```

Comment:
Can arrays be of the same shape, but not the same size?  If not (my intuition), I'd probably get rid of the size check (or reorder them) because if I got that message, it would imply to me that it would be OK to compare a 10x10 array to a 100 array, but then as soon as I tried it, I'd get the shape error message ("D'oh!  Bart!")

Reply:
Good point Brad. I'll tackle that in a follow-up PR.